### ISSUE TITLE
Carthage build not working as suggested

### ISSUE THREAD

**Issue by whitespace-software** [couchbase/couchbase-lite-ios#1153]
I want to use Couchbase-Lite from Carthage. 

With a Cartfile (as documented in https://github.com/couchbase/couchbase-lite-ios) of 
```
github "couchbase/couchbase-lite-ios" "release/1.2.0"
```
I get
```
*** Skipped building couchbase-lite-ios due to the error:
Dependency "couchbase-lite-ios" has no shared framework schemes for any of the platforms: iOS
```

With a Cartfile of
```
github "couchbase/couchbase-lite-ios" "master"
```
I get
```
A shell task failed with exit code 66:
xcodebuild: error: Scheme CBForest-iOS-Carthage is not currently configured for the build action.
```
--------
* Version:   10.11.2
* Client OS: OSX
* Server:    N/A

**Comment by pasin** [comment]
From what I found:
1. `release/1.2.0` branch doesn't have a shared scheme to build iOS framework. We could try to fix this in 1.2.1.
2. `master` branch has wrong build target names in set in `CBForest-iOS-Carthage`.

**Comment by pasin** [comment]
Fixing build targets in `CBForest-iOS-Carthage' fixes the iOS build issue but there are still issues with tvOS and Mac platform.

tvOS:
```
*** Skipped building couchbase-lite-ios due to the error:
Dependency "couchbase-lite-ios" has no shared framework schemes for any of the platforms: tvOS
```

Mac:
```
=== BUILD TARGET CBL Mac Unit Tests OF PROJECT CouchbaseLite WITH CONFIGURATION Release ===

Check dependencies
Swift is unavailable for i386 on OS X; please remove i386 from ARCHS (currently 'x86_64 i386')

** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
A shell task failed with exit code 65:
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
```

**Comment by snej** [comment]
We already added a commit (7f33ceb50bf78af5d7491d2571764598dca17b67) to the release/1.2.0 branch (and updated the tag) to fix the CBForest-iOS-Carthage scheme. Was that a different problem?

**Comment by pasin** [comment]
Yes, it's different issue. 7f33ceb50bf78af5d7491d2571764598dca17b67 fixed `CBForest-iOS-Carthage` scheme (needed to be merged into dev/master) but release/1.2.0 is still missing shared scheme to build iOS framework.

**Comment by pasin** [comment]
@whitespace-software This has been fixed in `release/1.2.1` (not officially released yet) and `dev` branch.
